### PR TITLE
adding blueprint examples for remote inference

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[adding blueprint examples for remote inference]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/remote_inference_blueprints/open_ai_connector_completion_blueprint.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+```json
+{
+    "name": "OpenAI Connector",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/open_ai_connector_completion_blueprint.md:3]:
```suggestion
    "name": "OpenAI Completions Connector",
```

File: docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md [hunk 2]
```diff
@@ -0,0 +1,4169 @@
...
+              "Say this is a test"
+            ],
+            "embeddings": [
+              [
+                -0.76953125,
+                -0.12731934,
+                -0.52246094,
+                -1.2714844,
+                1.8076172,
+                0.13842773,
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md:65]: Let keep the first five embedding float values, then use `...` to repreent the others. It seems not helping too much by showing thousands of floats.

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    "service_name": "sagemaker"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "url": "https://runtime.sagemaker.${parameters.region}.amazonaws.com/endpoints/lmi-model-2023-06-24-01-35-32-275/invocations",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:26]: Delete this URL , this is just our internal Sagemaker endpiont. 
```suggestion
      "url": "<PLEASE ADD YOUR Sagemaker URL>",
```

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+      "request_body": "[\"${parameters.inputs}\"]"
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:27]: This request body is just from our test sagemaker embedding model. User should add their own body template.
```suggestion
      "request_body": "<PLEASE ADD YOUR REQUEST BODY>"
```

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+```
+
+#### Sample response
+```json
+{
+  "connector_id": "XU5UiokBpXT9icfOM0vt"
+}
+```
+
+### Corresponding Predict request example:
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:40]: As we can predict what's the model deployed to Sagemaker on user side, so we should remove the predict request example and sample response.

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 6]
```diff
@@ -0,0 +1,454 @@
+### Sagemaker connector blueprint example for embedding:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "sagemaker: embedding",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:6]:
```suggestion
  "name": "<YOUR MODEL NAME>",
```

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+  "description": "Test connector for Sagemaker embedding model",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:7]:
```suggestion
  "description": "<YOUR MODEL DESCRIPTION>",
```

File: docs/remote_inference_blueprints/open_ai_connector_chat_blueprint.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+### OpenAI connector blueprint example for chat:
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/open_ai_connector_chat_blueprint.md:1]: Explain that this example is from OpenAI doc https://platform.openai.com/docs/api-reference/chat?

Comment (ylwu-amzn) [reply docs/remote_inference_blueprints/open_ai_connector_chat_blueprint.md:1]: Same for other examples, we should use the exact same example from OpenAI and Cohere official document. And we should put their doc link here so community user can refer to .

File: docs/remote_inference_blueprints/sagemaker_connector_blueprint.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "url": "<PLEASE ADD YOUR Sagemaker MODEL ENDPOINT URL>",
+      "request_body": "<PLEASE ADD YOUR REQUEST BODY. Example: [\"${parameters.inputs}\"]>"
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:27]: How about just `"{parameters.inputs}"` ? Add `[` and `]` looks confusing as that's only for embedding generation , not a general one

Comment (dhrubo-os) [reply docs/remote_inference_blueprints/sagemaker_connector_blueprint.md:27]: You mean: `${parameters.inputs}` ?

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks a lot

**Review by rbhavna — APPROVED** [review]

**Pull request merged**